pkg/controller/network/ippool/ippool_controller.go

```diff
@@ -163,6 +171,27 @@ func (c *IPPoolController) ValidateCreate(obj runtime.Object) error {
 	return nil
 }
 
+func (c *IPPoolController) validateDefaultIPPool(p *networkv1alpha1.IPPool) error {
+	pools, err := c.kubesphereClient.NetworkV1alpha1().IPPools().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(
+			labels.Set{
+				networkv1alpha1.IPPoolDefaultLabel: "",
+			}).String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	poolLen := len(pools.Items)
+	if poolLen > 1 || poolLen == 0 {
```

Comment:
```suggestion
	if poolLen != 1 || pools.Items[0].Name != p.Name {
	    return nil
	}
```

Reply:
The purpose here is to ensure that there will be at least one default ippool, but when there is no default ippool in etcd, creating an ippool will not fail.

Reply:
yeah, the suggested change is equal to the code below but simpler

```
if poolLen > 1 || poolLen == 0 {
    return nil
} else if poolLen == 1 && pools.Items[0].Name != p.Name {
    return nil
}
```

Reply:
i see